docs/reference/query-dsl/has-parent-query.asciidoc

```diff
@@ -1,93 +1,120 @@
-        "has_parent" : {
-            "parent_type" : "blog",
-            "query" : {
-                "term" : {
-                    "tag" : "something"
+    "mappings": {
+        "properties" : {
+            "my-join-field" : {
+                "type" : "join",
+                "relations": {
+                    "parent": "child"
                 }
             }
         }
     }
 }
---------------------------------------------------
-// CONSOLE
 
-Note that the `has_parent` is a slow query compared to other queries in the
-query dsl due to the fact that it performs a join. The performance degrades
-as the number of matching parent documents increases. If you care about query
-performance you should not use this query. However if you do happen to use
-this query then use it as less as possible. Each `has_parent` query that gets
-added to a search request can increase query time significantly.
-
-[float]
-==== Scoring capabilities
+----
+// CONSOLE
+// TESTSETUP
 
-The `has_parent` also has scoring support. The default is `false` which
-ignores the score from the parent document. The score is in this
-case equal to the boost on the `has_parent` query (Defaults to 1). If
-the score is set to `true`, then the score of the matching parent
-document is aggregated into the child documents belonging to the
-matching parent document. The score mode can be specified with the
-`score` field inside the `has_parent` query:
+[[has-parent-query-ex-query]]
+===== Example query
 
 [source,js]
---------------------------------------------------
+----
 GET /_search
 {
     "query": {
         "has_parent" : {
-            "parent_type" : "blog",
-            "score" : true,
+            "parent_type" : "parent",
             "query" : {
-                "term" : {
-                    "tag" : "something"
-                }
+                "match_all" : {}
             }
         }
     }
 }
---------------------------------------------------
+----
 // CONSOLE
 
-[float]
-==== Ignore Unmapped
+[[has-parent-top-level-params]]
+==== Top-level parameters for `has_parent`
+
+`parent_type`::
+(Required, string) Name of the parent relationship mapped for the
+<<parent-join,join>> field.
+
+`query`::
+(Required, query object) Query you wish to run on parent documents of the
+`parent_type` field. If a parent document matches the search, the query returns
+its child documents.
+
+`score`::
++
+--
+(Optional, boolean) Indicates whether the <<query-filter-context,relevance
+score>> of a matching parent document is aggregated into its child documents.
+Defaults to `false`.
 
-When set to `true` the `ignore_unmapped` option will ignore an unmapped `type`
-and will not match any documents for this query. This can be useful when
-querying multiple indexes which might have different mappings. When set to
-`false` (the default value) the query will throw an exception if the `type`
-is not mapped.
+If `false`, {es} ignores the relevance score of the parent document.
```

Comment:
may be also mention here, that in the case of `false`, the child docs' scores will be equal to `boost` of `has_parent` query which defaults to 1.

Reply:
Added with 4f290ab.